src/Documentation/grains/transactions.md

```diff
@@ -88,31 +116,54 @@ public class AccountGrain : Grain, IAccountGrain
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value += amount);
     }
 
     Task IAccountGrain.Withdrawal(uint ammount)
     {
-        this.balance.State.Value -= ammount;
-        this.balance.Save();
-        return Task.CompletedTask;
+        return this.balance.PerformUpdate(x => x.Value -= amount);
     }
 
     Task<uint> IAccountGrain.GetBalance()
     {
-        return Task.FromResult(this.balance.State.Value);
+        return this.balance.PerformRead(x => x.Value);
     }
-}
+}`
+```
+
+In the above example, the attribute `TransactionalState` is used to declare that the ‘balance’ constructor argument should be associated with a transactional state named “balance”.
+With this declaration, Orleans will wire up an `ITransactionalState` instance with a state loaded from the transactional state storage named "TransactionStore" (see Setup).
+The state can be modified via `PerformUpdate` or read via `PerformRead`.
+The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted upon completion of the grain call which created the transaction (`IATMGrain.Transfer` in the above examples).
+
+### Calling Transactions
+
+Transactional methods on a grain interface are called the same as any other grain call.
+
+``` csharp
+    IATMGrain atm = client.GetGrain<IATMGrain>(0);
+    Guid from = Guid.NewGuid();
+    Guid to = Guid.NewGuid();
+    await atm.Transfer(from, to, 100);
+    uint fromBalance = await client.GetGrain<IAccountGrain>(from).GetBalance();
+    uint toBalance = await client.GetGrain<IAccountGrain>(to).GetBalance();
 ```
 
-In the above example the attribute TransactionalState is used to declare that the ‘balance’ construction argument should be associated with a transactional state named “balance”.  With this declaration Orleans will wire up an ITransactionalState<Balance> instance with a state loaded from the default storage provider for the grain to use.  The state can be accessed and modified via the State property.  Any changes to the state need be recorded by calling ‘Save()’.  The transaction infrastructure will ensure that any such changes performed as part of a transaction, even among multiple grains distributed over an Orleans cluster, will either all be committed or reverted together.
+In the above calls, an atm grain is used to transfer 100 units of currency from one account to another.
+After the transfer is complete, both accounts are queried to get their current balance.
+The currency transfer as well as both account queries are performed as ACID transactions.
 
+As seen in the above example, transactions can return values within a task, like other grain calls, but upon call failure they will not throw application exceptions, but rather an `OrleansTransactionException` or `TimeoutException`.
```

Comment:
task -> `Task`?

Reply:
So we always wrap any user exceptions in transactional methods?

I feel uneasy about that, I'm not convinced it's the right thing to do from a user's perspective.

Reply:
The user exception question is another one for @jason-bragg 

And did you mean this...

> As seen in the above example, transactions can return values within a Task<T>, like other grain calls...

Reply:
Agreed, Task<T> is more clear.